cmake/OmimHelpers.cmake

```diff
@@ -3,6 +3,7 @@ set(OMIM_WARNING_FLAGS
   $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wall>
   $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wextra>
   $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wpedantic>
+  $<$<NOT:$<CXX_COMPILER_ID:MSVC,GNU>>:-Werror>
```

Comment:
It fails on my Big Sur Mac OS X:
```
error: unknown warning option '-Wno-unused-but-set-variable'; did you mean '-Wno-unused-const-variable'? [-Werror,-Wunknown-warning-option]
```

Let's avoid -Werror, for now, to avoid spending a lot of time with warning fixes instead of bug fixes or features implementation, and come back to it again in the future.